pyproject.toml

```diff
@@ -0,0 +1,11 @@
+[tool.ruff]
```

Comment:
This adds options for ruff but doesn't change the lint action workflow?

Reply:
It looks like there are some code changes that are due to Ruff formatting only? Given this is such a big PR I'd suggest staying on the old linting setup and we can switch to ruff later? (I'm interested to hear what you thought of ruff tho!)

Reply:
Not adding pyproject.toml for now